clang_tidy/clang_tidy.bzl

```
@@ -56,9 +56,12 @@ def _run_tidy(ctx, exe, flags, compilation_context, infile, discriminator):
 
 def _rule_sources(ctx):
     srcs = []
+    file_extensions = ["c", "cc", "cpp", "cxx", "c++", "C", "h", "hh", "hpp", "hxx", "inc", "inl", "H"]
     if hasattr(ctx.rule.attr, "srcs"):
         for src in ctx.rule.attr.srcs:
-            srcs += [src for src in src.files.to_list() if src.is_source]
+            for path in src.files.to_list():
+                if path.extension in file_extensions and path.is_source:
+                    srcs += [path] 
```

逻辑上没问题了，不过出于简洁，你可以这样改（当然这仅仅是建议）：

```suggestion
            srcs += [src for src in src.files.to_list() if src.is_source and path.extension in file_extensions]
```

这样修改后，回过头再对比一下最开始你没有修改的时候的样子，或许会发现原来这个问题如此简单